### DISCUSSION TITLE
@Profile에 대해서 설명해주세요.

### DISCUSSION THREAD

**Discussion by le2sky** [2024-woowacourse-study/level-interview#219] [Spring Core]
뭘까용

**Comment by J-I-H-O** [comment]
**<사용 이유>**
`@Profile` 어노테이션은 스프링 프레임워크에서 구성 요소를 구분함으로써 활성화된 프로파일에 따라 구성 요소가 다르게 등록되도록 합니다. 
즉, 특정 프로파일에서만 빈을 활성화하거나 비활성화할 수 있도록 해주는 기능을 제공합니다.

이를 통해 개발, 테스트, 운영 등 다양한 환경에 맞춰 애플리케이션의 동작을 다르게 구성할 수 있습니다.
특정 프로파일에서만 필요한 빈을 정의하여 불필요한 빈 로딩을 줄일 수 있으며, 테스트 환경에 맞춰 다양한 설정을 손쉽게 전환할 수 있어 테스트 작성과 유지보수가 용이합니다.

**Comment by le2sky** [comment]
`@Profile` 어노테이션을 사용하면 하나 이상의 프로필이 활성화되었을때 무엇을 등록할지 결정하는 힌트를 제공합니다.

```java
@Configuration
public class Config {

	@Bean
        @Profile("development") // profile이 development인 경우 등록
	public DataSource dataSource() {
		return new EmbeddedDatabaseBuilder()
			.setType(EmbeddedDatabaseType.HSQL)
			.addScript("classpath:com/bank/config/sql/schema.sql")
			.addScript("classpath:com/bank/config/sql/test-data.sql")
			.build();
	}
}
```
프로필은 ApplicationContext에게 알려주거나, spring.profiles.active 프로퍼티, web.xml 등을 이용해서 설정 가능합니다. 통합 테스트 환경에서는 `@ActiveProfiles`어노테이션을 이용해서 특정 프로필을 활성화하라고 ApplicationContext에게 알려 줄 수 있습니다.
